lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -63,6 +63,16 @@ def __init__(self, charm: "OpenSearchBaseCharm"):
 
     def _on_smtp_credentials_changed(self, event) -> None:
         """Creates secret containing key, value pairs for keystore"""
+        if not (deployment_desc := self.charm.opensearch_peer_cm.deployment_desc()):
+            logger.debug("Node not up yet. Deferring event.")
```

Comment:
for this and following occurrence in `_on_smtp_credentials_gone`
```suggestion
            logger.debug("Deployment not ready. Deferring event.")
```